generator/internal/dart/darttemplate_test.go

```diff
@@ -80,3 +81,82 @@ func TestAnnotateMethod(t *testing.T) {
 		t.Errorf("mismatched type, got=%q, want=%q", got, want)
 	}
 }
+
+func TestCalculateDependencies(t *testing.T) {
+	for _, test := range []struct {
+		name    string
+		imports []*dartImport
+		want    []string
+	}{
+		{name: "empty", imports: []*dartImport{}, want: []string{}},
+		{name: "dart import", imports: []*dartImport{typedDataImport}, want: []string{}},
+		{name: "package import", imports: []*dartImport{httpImport}, want: []string{"http"}},
+		{name: "dart and package imports", imports: []*dartImport{typedDataImport, httpImport}, want: []string{"http"}},
+		{name: "package imports", imports: []*dartImport{
+			httpImport,
+			{
+				Package:    "google_cloud_foo",
+				DartImport: "package:google_cloud_foo/foo.dart",
+			},
+		}, want: []string{"google_cloud_foo", "http"}},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			deps := map[string]*dartImport{}
+			for _, imp := range test.imports {
+				deps[imp.Package] = imp
+			}
+			gotFull := calculateDependencies(deps)
+
+			got := []string{}
+			for _, dep := range gotFull {
+				got = append(got, dep.Name)
+			}
+
+			if !reflect.DeepEqual(got, test.want) {
```

Comment:
we have been using `cmd.Diff()` for such tests. Any reason that does not work here? If so, it deserves a comment as it would be surprising.

Reply:
I wasn't aware of `Diff` :) I'll look to refactor this.